aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -510,4 +534,54 @@ Tensor& capped_random_cuda_(Tensor& self, int64_t to, Generator* gen) {
   return clamped_random_cuda_(self, 0, to, gen);
 }
 
+Tensor& normal_cuda_(Tensor& self, double mean, double std, Generator* gen) {
+  TORCH_CHECK(std > 0.0, "normal_ expects std > 0.0, but found std=", std);
+  auto iter = TensorIterator::nullary_op(self);
+  normal_kernel_cuda(*iter, mean, std, gen);
+  return self;
+}
+
+Tensor& normal_out_cuda(Tensor& output, const Tensor& mean_, double std, Generator* gen) {
+  auto mean = std::get<0>(expand_inplace(output, mean_.to(kCUDA)));
+  normal_cuda_(output, 0, std, gen);
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(output.scalar_type(), "normal_out_cuda_cadd", [&] {
+    at::add_out(output, output, mean, static_cast<scalar_t>(1));
+   });
+  return output;
+}
+
+Tensor& normal_out_cuda(Tensor& output, double mean, const Tensor& std_, Generator* gen) {
+  auto std = std::get<0>(expand_inplace(output, std_.to(kCUDA)));
```

Comment:
One thing I'm not sure about here, is why the explicit expand is necessary. Don't functions like `addcmul` broadcast automatically?

Reply:
Is perhaps the problem because you're directly calling the `at::native` function, and thus bypassing the logic that handles expansion?

Reply:
As above, also don't know if explicit expand is necessary. btw, for addcmul, this is what I want to do.
`output = mean_tensor + output * std`
`func: addcmul(Tensor self, Tensor tensor1, Tensor tensor2, *, Scalar value=1) -> Tensor` just addcmul returns a new tensor, so i could have done mean_tensor.addcmul(output, std). But then I would have to copy mean_tensor into output to return it? `addcmul_` does similar thing except inplace. And addcmul_out was the only API I found which would accumulate the `mean_tensor + output * std` in output. That's why I used it. Any comments?

Reply:
Updated with addcmul/addcmul_.